src/mappings/entities/mapAsset.ts

```diff
@@ -0,0 +1,436 @@
+};
+
+const handleRemoveDocuments = async (params: Record<string, any>) => {
+  const { ticker, ids } = params;
+  const asset = await getAsset(ticker);
+  asset.documents = asset.documents.filter(doc => !ids.includes(doc.id));
+  await asset.save();
+};
+
+const handleUpdateIdentifiers = async (params: Record<string, any>) => {
+  const { ticker, identifiers: newIdentifiers } = params;
+  const asset = await getAsset(ticker);
+  asset.identifiers = formatAssetIdentifiers(newIdentifiers);
+  await asset.save();
+};
+
+const handleMakeDivisible = async (params: Record<string, any>) => {
+  const { ticker } = params;
+  const asset = await getAsset(ticker);
+  asset.isDivisible = true;
+  await asset.save();
+};
+
+const handleIssue = async (params: Record<string, any>) => {
+  const { ticker, amount } = params;
+  const asset = await getAsset(ticker);
+  const formattedAmount = chainAmountToBigNumber(amount);
+  const assetOwner = await AssetHolder.get(`${asset.ownerDid}:${ticker}`);
+  assetOwner.amount = new BigNumber(assetOwner.amount).plus(formattedAmount).toString();
+  asset.totalSupply = new BigNumber(asset.totalSupply).plus(formattedAmount).toString();
+  await Promise.all([asset.save(), assetOwner.save()]);
+};
+
+const handleRedeem = async (params: Record<string, any>) => {
+  const { ticker, value: amount } = params;
+  const asset = await getAsset(ticker);
+  const formattedAmount = chainAmountToBigNumber(amount);
+  const assetOwner = await AssetHolder.get(`${asset.ownerDid}:${ticker}`);
+  assetOwner.amount = new BigNumber(assetOwner.amount).minus(formattedAmount).toString();
+  asset.totalSupply = new BigNumber(asset.totalSupply).minus(formattedAmount).toString();
+  await Promise.all([asset.save(), assetOwner.save()]);
+};
+
+const handleFreeze = async (params: Record<string, any>) => {
+  const { ticker } = params;
+  const asset = await getAsset(ticker);
+  asset.isFrozen = true;
+  await asset.save();
+};
+
+const handleUnfreeze = async (params: Record<string, any>) => {
+  const { ticker } = params;
+  const asset = await getAsset(ticker);
+  asset.isFrozen = false;
+  await asset.save();
+};
+
+const handleAcceptAssetOwnershipTransfer = async (params: Record<string, any>) => {
+  const { authId } = params;
+  const authorization = await getAuthorization(authId);
+  if (!authorization.ticker) return;
+  const asset = await getAsset(authorization.ticker);
+  asset.ownerDid = authorization.to;
+  await asset.save();
+  await AssetPendingOwnershipTransfer.remove(authId);
+};
+// #endregion
+
+// #region ModuleIdEnum.Identity
+const handleAddPendingOwnership = async (params: Record<string, any>, extrinsic: any) => {
+  const { target: targetData, authorizationData } = params;
+  const id = extrinsic.events[0].event.data[3].toString();
+  let ticker: string;
+  const from = extrinsic.events[0].event.data[0].toString();
+  const to = targetData.Identity.toString();
+  const type = Object.keys(authorizationData)[0];
```

Comment:
Move this type extraction near the top of the function and make sure it equals `TransferAssetOwnership` or `BecomeAgent`. 

Block 102 was failing because it was trying to process a `JoinIdentity` auth that was targeting an account.

Also they should be enum like values, I don't think we have them yet, but we should probably add Authorization types into `src/mappings/entities/common.ts`